code/datums/gamemodes/infestation.dm

```diff
@@ -33,6 +33,21 @@
 	for(var/i in GLOB.xeno_jelly_pod_turfs)
 		new /obj/structure/xeno/resin_jelly_pod(i, XENO_HIVE_NORMAL)
 
+	scale_weapons()
+
+/datum/game_mode/infestation/proc/scale_weapons(pop_override)
+	if(!isnum(pop_override))
+		pop_override = length(GLOB.alive_human_list)
+	var/item_ratio = floor(pop_override / INFESTATION_MARINE_PER_SCOPE)
+	for(var/obj/item/attachable/scope/mini AS in GLOB.vending_records["Attachments"])
+		if(mini == -1)
+			continue
+		mini = item_ratio
+	for(var/obj/item/weapon/gun AS in GLOB.vending_records["Marskman"])
+		if(gun == -1)
+			continue
+		gun = item_ratio
```

Comment:
uh, has this been tested?

Reply:
Didn't know of a way to test it on local, due to nuke war requiring a minimum number of xeno players to run

Reply:
you know you can just press the start button a 2nd time and it lets you bypass those restrictions, right?